libs/common/src/common/session_storage/redis.py

```diff
@@ -0,0 +1,41 @@
+from redis.asyncio import StrictRedis
+from typing import Optional
+
+from common.config import app
+from . import Session, SessionStorage
+
+class RedisSessionStorage(SessionStorage):
+    """
+    Redis session storage - Used to store and retrieve the session_id/identity header pair
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.redis_client = StrictRedis(
+            host=app.redis_hostname,
+            port=app.redis_port,
+            username=app.redis_username,
+            password=app.redis_password,
+            decode_responses=True,
+        )
+
+    async def retrieve(self, session_key: str) -> Optional[Session]:    
+        """Read the identity header from Redis using the session_id"""
+        try:
+            value = await self.redis_client.get(session_key)
+            if value:
+                return value
+            else:
+                return None
+        except Exception as e:
+            print(f"Error reading from Redis: {e}")
```

Comment:
Lets allow the error to propagate and be free! 
If this fails, we shouldn't keep processing the request anyway.
I don't think we need to tell the user about something in particular other than "whoops, this failed"